src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -2856,6 +2856,19 @@ public static unsafe void ConstructorsTest()
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)] // These tests assume that UTF8 is the default encoding
+        public static unsafe void ConstructorsTest_InvalidUTF8()
+        {
+            byte[] invalidUTF8Bytes = new byte[] { (byte)'A', (byte)'B', 0xEE, (byte)'C', (byte)'D', 0 };
+
+            fixed (byte* pBytes = invalidUTF8Bytes)
+            {
+                Assert.Equal("AB\ufffdCD", new String((sbyte*)pBytes));
+                Assert.Equal("B\ufffdC", new String((sbyte*)pBytes, 1, 3));
```

Comment:
- This behavior matches `Encoding.UTF8` behavior
- This behavior matches of Windows `MultiByteToWideChar` behavior that is used to implement these constructors on Windows (invalid characters are replaced with \uffffd without `MB_ERR_INVALID_CHARS` set)
- Mono throws ExecutionEngineException for the single parameter constructor (looks like a bug), and behaves as expected for the second case.